core/server/api_container/server/startosis_engine/startosis_packages/git_package_content_provider/git_package_content_provider_test.go

```diff
@@ -211,10 +236,37 @@ func TestGitPackageProvider_FailsForNonExistentPackage(t *testing.T) {
 	provider := NewGitPackageContentProvider(oackageDir, packageTmpDir, githubAuthTokenFilePath.Name(), nil)
 	nonExistentModulePath := "github.com/kurtosis-tech/non-existent-startosis-load/sample.star"
 
-	_, err = provider.GetModuleContents(nonExistentModulePath)
+	nonExistentModuleAbsoluteLocator := startosis_packages.NewPackageAbsoluteLocator(nonExistentModulePath, defaultMainBranch)
+
+	_, err = provider.GetModuleContents(nonExistentModuleAbsoluteLocator)
 	require.NotNil(t, err)
 }
 
+func TestGitPackageProvider_GetContentFromAbsoluteLocatorWithCommit(t *testing.T) {
+	packageDir, err := os.MkdirTemp("", packagesDirRelPath)
+	require.Nil(t, err)
+	defer os.RemoveAll(packageDir)
+	packageTmpDir, err := os.MkdirTemp("", repositoriesTmpDirRelPath)
+	require.Nil(t, err)
+	defer os.RemoveAll(packageTmpDir)
+	githubAuthDir, err := os.MkdirTemp("", githubAuthDirRelPath)
+	require.Nil(t, err)
+	githubAuthTokenFilePath, err := os.CreateTemp(githubAuthDir, githubAuthTokenFilename)
+	require.Nil(t, err)
+	defer os.RemoveAll(githubAuthDir)
+
+	provider := NewGitPackageContentProvider(packageDir, packageTmpDir, githubAuthTokenFilePath.Name(), nil)
+
+	absoluteLocatorStr := "github.com/kurtosis-tech/ethereum-package/src/package_io/input_parser.star"
```

Comment:
It's not great that we have tests that rely on external resources, I guess because the absolute locator reference to a pinned commit this is fine, but if you decide to update the test to account for my comment below, might have the reference the main branch which is susceptible to change. I wonder if we can mock git repos somehow or maybe refer to a package in this repo? but maybe thats for a separate change, I'll let you decide whats best here

Reply:
Yes it's not the best, now I'm using an archived repository to handle this problem, but it's not the final solution.

Doing the refactor is bigger than this change and probably out of the scope of this issue and I don't want to block it in order to unblock the user, so I created this ticket that we can prioritize later> https://github.com/kurtosis-tech/kurtosis/issues/2310